lib/config.js

```diff
@@ -284,3 +285,31 @@ function filterKeys (obj, keys) {
   for (const k in obj) { if (keys.indexOf(k) > 0) res[k] = obj[k] }
   return res
 }
+
+function mergeProperties (propertyKey, firstObject, secondObject) {
+  var propertyValue = firstObject[propertyKey]
+
+  if (secondObject === undefined || secondObject[propertyKey] === undefined) {
+    return firstObject[propertyKey]
+  } else if (typeof propertyValue === 'object') {
+    return mergeNestedObjects(firstObject[propertyKey], secondObject[propertyKey])
+  }
+
+  return secondObject[propertyKey]
+}
+
+function mergeNestedObjects (firstObject, secondObject) {
+  var finalObject = {}
+
+  // Merge first object and its properties.
+  for (var propertyKey in firstObject) {
+    finalObject[propertyKey] = mergeProperties(propertyKey, firstObject, secondObject)
+  }
+
+  // Merge second object and its properties.
+  for (var propertyKey2 in secondObject) {
+    finalObject[propertyKey2] = mergeProperties(propertyKey2, secondObject, firstObject)
+  }
+
+  return finalObject
+}
```

Comment:
So something like [_.merge](https://lodash.com/docs/#merge) might be a better approach here...